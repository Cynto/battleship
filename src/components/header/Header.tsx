import React from 'react';
import './header.css';
import { Link } from 'react-router-dom';
function Header() {
  return (
    <header data-testid="header" className="header">
      <Link to="/">
        <h1>Battleship</h1>
      </Link>
    </header>
  );
}

export default Header;
