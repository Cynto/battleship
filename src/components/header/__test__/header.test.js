import React from 'react';
import ReactDOM from 'react-dom';
import Header from '../Header';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { HashRouter as Router } from 'react-router-dom';

describe('Start page tests', () => {
  afterEach(cleanup);
  test('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Router>
        <Header />
      </Router>,
      div,
    );
  });
  test('Renders main container correctly', () => {
    const { getByTestId } = render(
      <Router>
        <Header />
      </Router>,
    );
    expect(getByTestId('header')).toHaveClass('header');
  });
  test('Matches snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Header />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
