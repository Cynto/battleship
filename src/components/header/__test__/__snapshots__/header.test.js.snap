// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start page tests Matches snapshot 1`] = `
<header
  className="header"
  data-testid="header"
>
  <a
    href="#/"
    onClick={[Function]}
  >
    <h1>
      Battleship
    </h1>
  </a>
</header>
`;
